### PR TITLE
bpo-45548: Have Modules/Setup only try to compile _math once

### PR DESCRIPTION
**Pull request by brettcannon**
https://bugs.python.org/issue45548

### CODE REVIEW

File: Modules/Setup [hunk 1]
```diff
@@ -172,7 +172,7 @@ time timemodule.c
 #audioop audioop.c
 #binascii binascii.c
 #cmath cmathmodule.c _math.c  # -lm
-#math mathmodule.c _math.c  # -lm
+#math mathmodule.c # _math.c  # -lm  # Only one of cmath/math needs to compile _math.c
```
Comment (tiran) [inline Modules/Setup:175]: I don't think that will work with ``*shared**`` mode.

Comment (brettcannon) [reply Modules/Setup:175]: You're right.
```
/usr/bin/ld: Modules/cmathmodule.o: relocation R_X86_64_32 against `.data' can not be used when making a shared object; recompile with -fPIC`
```

Comment (brettcannon) [reply Modules/Setup:175]: Your idea of making the functions static might actually be necessary to solve this. Not sure how `_math.h` influences anything as I haven't looked to see if it's included anywhere else.

Comment (tiran) [reply Modules/Setup:175]: The ``relocation R_X86_64_32 against `.data'`` issue is a different problem. You are trying to use an ``.o`` file from a static build in a shared module. You must ``make clean`` every time you modify ``Modules/Setup``. ``ccache`` helps a **lot** here.

**Pull request closed without merging**